### PR TITLE
I65 dockerize app

### PR DESCRIPTION
**Pull request by aprilrieger**
Ref ticket: https://github.com/scientist-softserv/cal-state-hyrax/issues/65

Dockerizes the application to include
- Removes Dockerfile.base and combines in Dockerfile
- Adds solr Dockerfile
- Adds docker-compose file
- Adds solr cloud
- Adds initialize app
- Add db-see-migrations
- Updates README

Testing Instructions:
- clone the app
- cd folder
- docker compose build
- dory up
- docker compose up
- navigate to `calstate.test`
- login as `kcloud@calstate.edu` and typical testing password.

Screenshots:
Screenshot 2023-01-28 at 15 30 21
Screenshot 2023-01-28 at 15 30 05

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
-DOCKER_PORT=3001:3001
-HOST=0.0.0.0
-
-HANDLE_SERVER=54.244.77.182:8000
-HS_ADMIN=300:0.NA/20.500.12680
-HS_PRIVATE_KEY=admpriv.pem
-HS_PREFIX=20.500.12680
+# DOCKER_PORT=3001:3001
+# HOST=0.0.0.0
+# HANDLE_SERVER=54.244.77.182:8000
```
Comment (labradford) [inline .env:5]: I think we need to uncomment these as they are needed for a gem they are using
https://github.com/csuscholarworks/handle-system

Comment (labradford) [reply .env:5]: also, not the biggest deal but we should keep the vars alphabetized

File: config/environments/development.rb [hunk 2]
```diff
@@ -53,5 +53,5 @@
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
```
Comment (labradford) [inline config/environments/development.rb:56]: uncomment for macs that aren't m1

File: solr/conf/schema.xml [hunk 3]
```diff
@@ -308,7 +308,6 @@
       <analyzer>
         <tokenizer class="solr.StandardTokenizerFactory"/>
         <filter class="solr.StopFilterFactory" ignoreCase="true" words="stopwords.txt"/>
-        <filter class="solr.StandardFilterFactory"/>
```
Comment (labradford) [inline solr/conf/schema.xml:311]: are we sure we want to change the solr schema?

**Review by labradford — CHANGES_REQUESTED** [review]
Do we want to add the staging deploy.yml file? 
Do we need to change the solr schema file?
I think we need to uncomment a few env vars and the file watcher thing

**Review by labradford — COMMENTED** [review]
FYI - I'm getting issues with the rails version locally

File: solr/conf/schema.xml (hunk 3 above)
Comment (aprilrieger) [reply solr/conf/schema.xml:311]: All the changes to the schema.xml files were due to the upgrade to solr8 service. Each of them were no longer supported in solr 8.

**Comment by aprilrieger** [comment]
> 

Can you please expand on this and let me know the specific issues you are facing?

**Comment by labradford** [comment]
> FYI - I'm getting issues with the rails version locally

This might be because I tried to fix the merge conflicts locally. I'll update you tomorrow

**Pull request closed without merging**